web/hooks/use-state-check-equality.ts

```diff
@@ -8,8 +8,9 @@ export const useStateCheckEquality = <T>(initialState: T) => {
   stateRef.current = state
 
   const checkSetState = useMemo(
-    () => (newState: T) => {
+    () => (next: SetStateAction<T>) => {
       const state = stateRef.current
+      const newState = next instanceof Function ? next(state) : next
```

Comment:
Please never use this hook to try to set state which is a function, but not an updater function, or I will cry.

Reply:
🙃